### PR TITLE
Add a ButtonCreator component, update to UI

### LINKED ISSUE #27: Improve Queue UI
- Sticky Join + Leave queue (possibly use GameUi)
- improve ui with button

### PR DESCRIPTION
**Pull request by KinTatHo**
JoinQueue and LeaveQueue now has buttons, and will not be overshadowed by walls

### CODE REVIEW

**Comment by KinTatHo** [comment]
Create a username popup

Users are now prompted to enter their username before joining the game

File: client/package-lock.json [hunk 1]
```diff
@@ -1352,7 +1352,6 @@
 			"version": "0.12.7",
 			"resolved": "https://registry.npmjs.org/path/-/path-0.12.7.tgz",
 			"integrity": "sha512-aXXC6s+1w7otVF9UletFkFcDsJeO7lSZBPUQhtb5O0xJe8LtYhj/GxldoL09bBj9+ZmE2hNoHqQSFMN5fikh4Q==",
-			"dev": true,
```
Comment (winson8222) [inline client/package-lock.json:1355]: How come this is added suddenly.

File: client/src/components/ButtonCreator.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      .setInteractive(
+        new Phaser.Geom.Rectangle(buttonX, buttonY, buttonWidth, buttonHeight),
+        Phaser.Geom.Rectangle.Contains
+      )
+      .setScrollFactor(0);
+
+    let buttonText = scene.add
+      .text(buttonX + buttonWidth / 2, buttonY + buttonHeight / 2, text, {
+        color: "#ffffff",
+        fontSize: "11px",
```
Comment (winson8222) [inline client/src/components/ButtonCreator.tsx:31]: If can cutomise the button size lets make font size customisabale also. This is a really useful creation function though.

File: client/src/scenes/Game.ts [hunk 3]
```diff
@@ -299,6 +328,7 @@ export default class Game extends Phaser.Scene {
         text,
         popupStyle
       )
+      .setScrollFactor(0)
```
Comment (winson8222) [inline client/src/scenes/Game.ts:315]: What is this change for?

**Review by winson8222 — CHANGES_REQUESTED** [review]
Very nice looking buttons, maybe can disable player movement when entering the username.

**Review by winson8222 — APPROVED** [review]

**Pull request merged**